src/charm.py

```diff
@@ -107,6 +156,65 @@ def _on_install(self, event: InstallEvent) -> None:
         """Handle the `on_install` event."""
         self.unit.status = WaitingStatus("Waiting for Pebble")
 
+    def _on_model_config_changed(self, event: HookEvent) -> None:
+        """Handle the `on_config_changed` event."""
+        self.refresh_cached_s3_credentials(event)
+
+    def _on_peer_relation_changed(self, event: RelationChangedEvent):
```

Comment:
If they don't do anything, best to avoid their definition to keep code simple, clean and concise